### PR TITLE
Add name to connection API

### PR DESCRIPTION
**Pull request by JPry**
### Changes proposed in this Pull Request:

This updates the `/mc/connection` API endpoint to return the name and domain as part of the response.

### Detailed test instructions:

1. Make a request to `/mc/connection` on the site.
2. Observe that the response now includes `name` and `domain` parameters.

### CODE REVIEW

File: src/API/Site/Controllers/MerchantCenter/AccountController.php [hunk 1]
```diff
@@ -76,7 +76,7 @@ public function __construct( ContainerInterface $container ) {
 		parent::__construct( $container->get( RESTServer::class ) );
 		$this->middleware    = $container->get( Middleware::class );
 		$this->merchant      = $container->get( Merchant::class );
-		$this->account_state = $container->get( MerchantAccountState::class );
+		$this->account_state = $container->get( AccountState::class );
```
Comment (mikkamp) [inline src/API/Site/Controllers/MerchantCenter/AccountController.php]: I don't think we should be changing this here (and I assume is the reason why the tests are stating that AccountState is not being managed by the container). Fails at the same location when sending the request.

**Review by mikkamp — COMMENTED** [review]
I was able to get it working after changing the get call for the container to `MerchantAccountStatus`. Once that was changed I get the expected response.

However for a site where I'm updating the plugin the account name will remain `null` until I reconnect my site. Would it be better in this case to fetch the account name from the Google API once? Kind of similar to what we did with the [ads currency](https://github.com/woocommerce/google-listings-and-ads/blob/1.8.0/src/API/Google/Proxy.php#L469). Although that makes me wonder, why are we storing the account name locally if it can be changed in Google (ad currency can't be changed in Google)? Based on the code here we won't catch any account name changes that have been done in Google.

Renaming to `AccountState::STEP_DONE` looks simpler. But do we want to do the same for the `AdsAccountState` either in this PR or a follow up PR?

File: src/API/Site/Controllers/MerchantCenter/AccountController.php (hunk 1 above)
Comment (JPry) [reply src/API/Site/Controllers/MerchantCenter/AccountController.php]: @mikkamp You're right, I was a little over-zealous in my find/replace. I've fixed that.

**Comment by JPry** [comment]
@mikkamp Thanks for the review 🙂 

> However for a site where I'm updating the plugin the account name will remain null until I reconnect my site. Would it be better in this case to fetch the account name from the Google API once? Kind of similar to what we did with the ads currency. Although that makes me wonder, why are we storing the account name locally if it can be changed in Google (ad currency can't be changed in Google)? Based on the code here we won't catch any account name changes that have been done in Google.

You're right, it would be better to handle this a bit more smoothly. I'm going to do some more reworking to take those other things into account.

**Pull request closed without merging**